### ISSUE TITLE
Add notion of internal/external to options

### ISSUE THREAD

**Issue by leshi** [w3c/webauthn#213]
Add notion of internal/external to options so that Caller can indicate that it only wants to talk to internal authenticators.

Example:

```
dictionary CredentialOptions {
    unsigned long                       timeoutSeconds;
    sequence < CredentialDescription >  excludeList;
    WebAuthnExtensions                  extensions;
    Attachment                          attachment;
};

enum Attachment {
    "INTERNAL",
    "EXTERNAL"
}
```

**Comment by jcjones** [comment]
Rather than this enumeration, perhaps we should look for something a bit richer, such as what [Media Capture and Streams API does for getUserMedia](https://w3c.github.io/mediacapture-main/#dom-mediadevices-getusermedia). (Moz docs:  https://developer.mozilla.org/en-US/docs/Web/API/MediaDevices/getUserMedia)

**Comment by rlin1** [comment]
Note: internal/external and second-factor (only)/first-factor authentictors are different things.

**Comment by yaronf** [comment]
And then maybe rename ``Attachment`` to ``AuthenticatorAttachment`` or even ``AuthenticatorFilter``.

**Comment by equalsJeffH** [comment]
to build on @rlin1 's comment, there are perhaps more authnr attributes that may be of interest than just "internal / external", and thus a single 1-dimensional enum may prove insufficiently expressive.  However, we may not need the expressability of the rich "[constrainable patten](https://w3c.github.io/mediacapture-main/#constrainable-interface)" from "Media Capture and Streams", that @jcjones points to.

**Comment by vijaybh** [comment]
I agree with @equalsJeffH - I wonder if we could just adopt this suggestion as-is and add more members to the Options dictionary in future if strong use cases come up. It's not clear to me that simply adding another layer of nesting (e.g. a Constraints structure with all these dimensions inside it) would be superior enough to justify the overhead.

**Comment by leshi** [comment]
I like this simple structure.  Though I think the suggested naming was `PLATFORM` instead of 'internal' and `CROSS_PLATFORM` instead of 'external'.

**Comment by leshi** [comment]
Took an initial stab at this in #243

**Comment by apowers313** [comment]
Section 5.1.1 says:

> Prompt the user for consent to create a new credential. The prompt for obtaining this consent is shown by the authenticator if it has its own output capability, or by the user agent otherwise. 

As long as we are doing the modification above, do we need a "hasPrompt" option too? Or do you guys have another way in mind of figuring out if an authnr can prompt users?

**Comment by equalsJeffH** [comment]
some quick comments:
* the notion of "attachment" is arguably a characteristic of authenticators (authnrs) rather than  credentials
* another authnr characteristic is the "2nd factor only" one, for which we have not come to firm conclusions. here's the latest msg in that thread: https://lists.w3.org/Archives/Public/public-webauthn/2016Jun/0086.html
* i prefer the term "roaming" to "cross-platform", but am fine with "platform" for bound/embedded/internal authnrs

I suggest we punt this issue from WD-03 (to WD-04 or CR) for now

**Comment by equalsJeffH** [comment]
> I suggest we punt this issue from WD-03 (to WD-04 or CR) for now

i.e. in order to get WD-03 published sooner rather than later

**Comment by leshi** [comment]
This is now closed with PR #243
